### PR TITLE
mudlet: Fix build on darwin

### PR DESCRIPTION
**Pull request by felixalbrigtsen**
## Description of changes

Fixed mudlet on Darwin by adding the dependency for AppKit and adapting the installPhase.
AppKit requires a patch because it is only included when the updater is enabled, which makes our configuration impossible.
No changes have been made to the linux build.

Result of `nixpkgs-review` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mudlet</li>
  </ul>
</details>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/games/mudlet/darwin-AppKit.patch
Comment (wegank) [inline pkgs/games/mudlet/darwin-AppKit.patch:6]: It seems there's already a conditional "-framework AppKit"?

Comment (felixalbrigtsen) [reply pkgs/games/mudlet/darwin-AppKit.patch:6]: Yes, it should be visible in the patch#L7, but [the original CMakeLists.txt](https://github.com/Mudlet/Mudlet/blob/3bd76f24c81226338290e3295ff4833083b8bfd3/src/CMakeLists.txt#L566) only links with AppKit when the self-updater is enabled.
Some parts of AppKit, namely the _NSAccessibility* section is referenced a few places in the app, like https://github.com/Mudlet/Mudlet/blob/3bd76f24c81226338290e3295ff4833083b8bfd3/src/AnnouncerMac.mm.

This might be more of an upstream problem? I have not tried building it outside of nix, but it looks like an error in the original cmake files, since it will not build without the updater.

Comment (wegank) [reply pkgs/games/mudlet/darwin-AppKit.patch:6]: Would you like to file a PR upstream dropping the conditions on `USE_UPDATER`? I can do it for you later this afternoon if you wish.

Comment (felixalbrigtsen) [reply pkgs/games/mudlet/darwin-AppKit.patch:6]: Sure, I can do it, but thanks anyways!

**Comment by pstn** [comment]
Would you like to add yourself as a maintainer so we can better test mac builds of this in the future?

**Comment by felixalbrigtsen** [comment]
I have added myself as maintainer, and believe it is all good to go now.

**Comment by wegank** [comment]
@ofborg build mudlet mudlet.passthru.tests

**Review by wegank — APPROVED** [review]
Tested locally on both aarch64-darwin and x86_64-linux.

**Review by pstn — APPROVED** [review]

File: pkgs/games/mudlet/default.nix
Comment (felixalbrigtsen) [reply pkgs/games/mudlet/default.nix]: I believe Wegank already fixed this in https://github.com/NixOS/nixpkgs/pull/281657/commits/94b3f101bcc43e5bbd82eebb194c7db6d27a760f

**Review by pstn — APPROVED** [review]

File: pkgs/games/mudlet/default.nix
Comment (pstn) [reply pkgs/games/mudlet/default.nix]: I think this one just didn't get resolved because I was commenting on a line that wasn't there yet. Looks good to me know though.

**Comment by felixalbrigtsen** [comment]
@wegank, is there anything else I need to do before merge?

**Pull request merged**
